RestaurantReservationSystem/EntityServices/CustomerService.cs

```diff
@@ -0,0 +1,111 @@
+﻿using RestaurantReservation.Db.Models;
+using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.Services.Interfaces;
+using RestaurantReservation.Db.Repositories.Interfaces;
+using RestaurantReservationSystem.Constants;
+
+public class CustomerService : ICustomerService 
+{
+    private readonly ICustomerRepository _customerRepository;
+
+    public CustomerService(ICustomerRepository customerRepository)
+    {
+        _customerRepository = customerRepository;
+    }
+
+    public async Task AddCustomerAsync(string firstName, string lastName, string email, string phoneNumber)
+    {
+        var customer = new Customer
+        {
+            FirstName = firstName,
+            LastName = lastName,
+            Email = email,
+            PhoneNumber = phoneNumber
+        };
+
+        try
+        {
+            await _customerRepository.AddAsync(customer);
+        }
+        catch (DbUpdateException ex)
+        {
+            throw new InvalidOperationException(DefaultErrorMessages.AddFailed, ex);
+        }
+    }
+
+    public async Task GetAllCustomersAsync()
+    {
+        try
+        {
+            var all = await _customerRepository.GetAllAsync();
+            foreach (var customer in all)
+            {
+                Console.WriteLine($"[Customer] {customer.FirstName} - {customer.LastName}," +
```

Comment:
we should be returning the list and caller can print it or do whatever they want with  it.

Reply:
same comment for the other services.